earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
+
+    # Install extra debian packages we will need to support plugins.
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        graphviz
+
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    RUN poetry config installer.max-workers 10
+
+    # Copy our dependencies.
+    COPY pyproject.toml .
+    COPY poetry.lock .
+    COPY start.sh .
+
+    # Install it all with poetry
+    RUN poetry install --only main --no-root
+
+    # Trust directory, required for git >= 2.35.2
+    RUN git config --global --add safe.directory /docs &&\
+        git config --global --add safe.directory /site
+
+    # Carry the Kroki Docker Compose with this image.
+    WORKDIR /kroki
+    COPY docker-compose.kroki.yml ./docker-compose.yml
+
+    # Set working directory
+    WORKDIR /docs
+
+    # Expose MkDocs development server port
+    EXPOSE 8000
+
+    # Start development server by default
+    ENTRYPOINT ["/poetry/start.sh"]
+    CMD ["serve", "--dev-addr=0.0.0.0:8000"]
+
+    # Finish building the container, and publish to the registry
+    IF [ "$registry" = "" ]
+        ARG registry_final=$registry
+    ELSE
+        ARG registry_final=${registry}/
+    END
+
+    SAVE ARTIFACT /kroki/docker-compose.yml /kroki/docker-compose.yml
```

Comment:
This seems slightly redundant. The file wasn't modified, why save it as an artifact? Do we intend to mangle it in the future?

Reply:
This could be my misunderstanding.  I thought one needed to `SAVE ARTIFACT` to `COPY` it into another target?

Reply:
Nevermind, I didn't understand what you were doing when I first made this comment. I see the reason this is needed since it's being pulled into a UDC (where `COPY` wouldn't work the way we need it).